compiler/qsc_eval/src/lib.rs

```diff
@@ -386,7 +387,7 @@ impl Env {
 
 #[derive(Default)]
 struct Scope {
-    bindings: HashMap<NodeId, Variable>,
+    bindings: FxHashMap<NodeId, Variable>,
```

Comment:
Why a hash map instead of an `IndexMap` in the first place? Looks like in a lot of places we use `IndexMap` for `NodeId`s and other sequentially generated integers.

Reply:
See other comment